dvc/ignore.py

```diff
@@ -115,16 +115,20 @@ def __call__(self, root, dirs, files):
 
 
 class CleanTree(BaseTree):
-    def __init__(self, tree):
+    def __init__(self, tree, tree_root=None):
         self.tree = tree
+        if tree_root:
+            self._tree_root = tree_root
+        else:
+            self._tree_root = self.tree.tree_root
 
     @cached_property
     def dvcignore(self):
         return DvcIgnoreFilter(self.tree)
```

Comment:
Hm, so dvcignore is still initialized with the host repo's root, right?

Reply:
I guess it is an interesting question overall: should host's dvcignore affect subrepo? IIRC, git submodules are not affected by the host's .gitignore. CC @pared

Reply:
Hmm, looks like you're right. `DvcIgnoreFilter.__init__` should probably also take a parameter for the cleantree root, since it will be different than the git tree root for subrepos.

And I don't think that git submodules are affected by the parent repo's gitignore.

Reply:
Separate issue (#3894) has been filed for this bug, it seems like this could potentially be causing other issues in DVC besides dvcignore, where we incorrectly use the host repo DVC root instead of subrepo DVC root whenever we are using brancher.